### PR TITLE
Various improvements to CMake find modules

### PR DESCRIPTION
**Pull request by Idclip**
This PR is a partial implementation of #654 

 - Removing unnecessary DLL methods
 - Adding better lib type detection on UNIX
 - Added GNU path support
 - Added required defines for static TBB on Windows
 - Added required defines for static IlmBase and OpenEXR on Windows
 - Improved handling of finding static builds of OpenVDB

Signed-off-by: Nick Avramoussis <4256455+Idclip@users.noreply.github.com>

### CODE REVIEW

File: cmake/FindIlmBase.cmake [hunk 1]
```diff
@@ -328,11 +303,37 @@ if(IlmBase_FOUND)
...
+    endif()
+
+    set(IlmBase_${COMPONENT}_DEFINITIONS)
+
+    # Add the OPENEXR_DLL define if the library is not static on WIN32
+    if(WIN32)
+      if(NOT ILMBASE_${COMPONENT}_LIB_TYPE STREQUAL STATIC)
+        list(APPEND IlmBase_${COMPONENT}_DEFINITIONS -DOPENEXR_DLL)
+      endif()
+    endif()
```
Comment (Idclip) [inline cmake/FindIlmBase.cmake:329]: Added explicit define for `OPENEXR_DLL` (see PlatformConfig.h)

File: cmake/FindTBB.cmake [hunk 2]
```diff
@@ -192,55 +193,27 @@ list(APPEND _TBB_LIBRARYDIR_SEARCH_DIRS
...
 else()
-  if(${CMAKE_CXX_COMPILER_ID} STREQUAL GNU)
-    if(TBB_MATCH_COMPILER_VERSION)
-      string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${CMAKE_CXX_COMPILER_VERSION})
-      list(GET GCC_VERSION_COMPONENTS 0 GCC_MAJOR)
-      list(GET GCC_VERSION_COMPONENTS 1 GCC_MINOR)
-      list(INSERT TBB_PATH_SUFFIXES 0 lib/intel64/gcc${GCC_MAJOR}.${GCC_MINOR})
-    else()
-      list(INSERT TBB_PATH_SUFFIXES 0 lib/intel64/gcc4.4)
-    endif()
```
Comment (Idclip) [inline cmake/FindTBB.cmake:228]: Removed all this in favour of gnu paths

**Comment by Idclip** [comment]
@danrbailey the Houdini CI builds seems to have immediately completed without any tests. Is that correct?

File: cmake/FindBlosc.cmake [hunk 3]
```diff
@@ -161,27 +162,32 @@ list(APPEND _BLOSC_LIBRARYDIR_SEARCH_DIRS
...
 endif()
 
-set(BLOSC_PATH_SUFFIXES
-  lib64
-  lib
-)
+# libblosc is the name of the blosc static lib on windows
 
-find_library(Blosc_LIBRARY blosc
+find_library(Blosc_LIBRARY blosc libblosc
```
Comment (Idclip) [inline cmake/FindBlosc.cmake:181]: also try and find libblosc

**Comment by danrbailey** [comment]
> @danrbailey the Houdini CI builds seems to have immediately completed without any tests. Is that correct?

Correct. The way things are currently setup, there's no way to run Houdini builds using GitHub Actions. Our Azure CI is here for that use case until we've transitioned to a workflow that supports this. I'll let the CI stabilize a little due to the recent changes and disk space issues before we look to change this.

**Pull request merged**